src/ascii_dialog/dialog.py

```diff
@@ -0,0 +1,383 @@
+        options =  self.datasetOptions()
+        self.col_editor = ColEditor(self.colcount_entry.value(), options)
+        self.dataset_combobox.currentTextChanged.connect(self.changeDatasetType)
+        self.col_editor.column_changed.connect(self.updateColumn)
+
+        ## Data Table
+
+        self.table = QTableWidget()
+        self.table.show()
+        # Make the table readonly
+        self.table.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
+        # The table's width will always resize to fit the amount of space it has.
+        self.table.setSizeAdjustPolicy(QAbstractScrollArea.SizeAdjustPolicy.AdjustToContents)
+        # Add the context menu
+        self.table.setContextMenuPolicy(Qt.ContextMenuPolicy.CustomContextMenu)
+        self.table.customContextMenuRequested.connect(self.showContextMenu)
+
+        # Warning Label
+        self.warning_label = WarningLabel(self.requiredMissing(), self.duplicateColumns())
+
+        self.layout = QVBoxLayout(self)
+
+        self.layout.addWidget(self.filename_label)
+        self.layout.addWidget(self.filename_chooser)
+        self.layout.addWidget(self.load_button)
+        self.layout.addLayout(self.dataset_layout)
+        self.layout.addLayout(self.sep_layout)
+        self.layout.addLayout(self.startline_layout)
+        self.layout.addLayout(self.colcount_layout)
+        self.layout.addWidget(self.col_editor)
+        self.layout.addWidget(self.table)
+        self.layout.addWidget(self.warning_label)
+
+    @property
+    def raw_csv(self) -> list[str] | None:
+        if self.current_filename is None:
+            return None
+        return self.files[self.current_filename]
+
+    @property
+    def rows_is_included(self) -> list[bool] | None:
+        if self.current_filename is None:
+            return None
+        return self.files_is_included[self.current_filename]
+
+
+    def splitLine(self, line: str) -> list[str]:
+        """Split a line in a CSV file based on which seperators the user has
+        selected on the widget.
+
+        """
+        expr = ''
+        for seperator, isenabled in self.seperators.items():
+            if isenabled:
+                if expr != r'':
+                    expr += r'|'
+                match seperator:
+                    case 'Comma':
+                        seperator_text = r','
+                    case 'Whitespace':
+                        seperator_text = r'\s+'
+                    case 'Tab':
+                        seperator_text = r'\t'
+                expr += seperator_text
+
+        return re.split(expr, line)
+
+    def attemptGuesses(self) -> None:
+        """Attempt to guess various parameters of the data to provide some
+        default values. Uses the guess.py module
+
+        """
+        split_csv = [self.splitLine(line.strip()) for line in self.raw_csv]
+
+        self.initial_starting_pos = guess_starting_position(split_csv)
+
+        guessed_colcount = guess_column_count(split_csv,
```

Comment:
Our standard is 120 chars per line so there is no need to split the lines so aggressively

Reply:
I see your point. I get my editor to split the lines automatically, and it looks like it was set to only 80 characters so I've bumped it up to 120.